### PR TITLE
Updated Nodes to work on ROS Eloquent with OpenCV 4

### PR DESCRIPTION
**Pull request by caelinsutch**
Fix implemented for #515 along with my working launch file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,6 @@
 *pyc
+cmake-build-debug/
```
Comment (SteveMacenski) [inline .gitignore]: Please remove your gitignores

File: image_proc/CMakeLists.txt [hunk 2]
```diff
@@ -14,213 +14,92 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rclcpp_components REQUIRED)
-find_package(rcutils REQUIRED)
 find_package(cv_bridge REQUIRED)
 find_package(image_transport REQUIRED)
 find_package(sensor_msgs REQUIRED)
-find_package(OpenCV 3 REQUIRED)
+find_package(OpenCV 4 REQUIRED)
```
Comment (SteveMacenski) [inline image_proc/CMakeLists.txt:20]: Dashing and Eloquent's target is still openCV3, but you're right that I believe Foxy will be opencv 4. We should deal with branching at then at least @JWhitleyWork

File: image_proc/CMakeLists.txt [hunk 3]
```diff
@@ -14,213 +14,92 @@ endif()
...
-  src/rectify.cpp)
+        src/rectify.cpp)
 target_compile_definitions(rectify
-  PRIVATE "COMPOSITION_BUILDING_DLL")
+        PRIVATE "COMPOSITION_BUILDING_DLL")
 
 ament_target_dependencies(rectify
-  ${dependencies}
-)
+        "image_geometry"
```
Comment (SteveMacenski) [inline image_proc/CMakeLists.txt]: Revert to use the dependencies variables. This is redundant.

File: image_proc/launch/image_proc.launch.py [hunk 4]
```diff
@@ -1,65 +1,98 @@
-# Copyright (c) 2020, Open Source Robotics Foundation, Inc.
-# All rights reserved.
+# Copyright 2018 Open Source Robotics Foundation, Inc.
```
Comment (SteveMacenski) [inline image_proc/launch/image_proc.launch.py]: You do not have permission to relicense OSRF's work. Revert.

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
I didn't look at the changes you made to the specific components yet. This is a _lot_ of changes. Please better describe what you did and why, or consider splitting into multiple PRs. 

I also am not sure why you modified so much of the cmakelists.

File: image_proc/src/rectify.cpp [hunk 5]
```diff
@@ -1,171 +1,169 @@
...
+        std::lock_guard<std::mutex> lock(connect_mutex_);
+
+        /*
+        *  SubscriberStatusCallback not yet implemented
+        */
+        /*if (pub_rect_.getNumSubscribers() == 0)
+          sub_camera_.shutdown();
+        else if (!sub_camera_)
+        {*/
+        RCLCPP_INFO(get_logger(), "Subscribe to topic: %s", image_topic.c_str());
```
Comment (SteveMacenski) [inline image_proc/src/rectify.cpp:113]: This is without a doubt going to fail linters. Please make sure you're in complience with `ament_cpplint` ament_uncrustify`, `ament_flake8`. We have spent alot of time to make the linters happy.

File: image_publisher/src/image_publisher.cpp [hunk 6]
```diff
@@ -124,7 +124,7 @@ void ImagePublisher::doWork()
   try {
     if (cap_.isOpened()) {
       if (!cap_.read(image_)) {
-        cap_.set(CV_CAP_PROP_POS_FRAMES, 0);
+        cap_.set(cv::CAP_PROP_POS_FRAMES , 0);
```
Comment (caelinsutch) [inline image_publisher/src/image_publisher.cpp:127]: Added a cv namespace for CV4

File: image_proc/launch/image_proc.launch.py [hunk 7]
```diff
@@ -30,36 +30,85 @@
 # ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
```
Comment (caelinsutch) [inline image_proc/launch/image_proc.launch.py:32]: Updated launch example with a more comprehensive version, taken from #457

File: image_proc/src/rectify.cpp (hunk 5 above)
Comment (caelinsutch) [reply image_proc/src/rectify.cpp:113]: How do you use those linters? I haven't had any experience with them.

File: image_publisher/CMakeLists.txt [hunk 8]
```diff
@@ -29,15 +29,9 @@ set( DEPENDENCIES
   cv_bridge
 )
 
-set(opencv_2_components core highgui)
-set(opencv_3_components core imgcodecs videoio)
+set(opencv_4_components core imgcodecs videoio)
```
Comment (caelinsutch) [inline image_publisher/CMakeLists.txt:32]: Default to just opencv4, may or may not be wanted @SteveMacenski

File: image_proc/src/rectify.cpp (hunk 5 above)
Comment (SteveMacenski) [reply image_proc/src/rectify.cpp:113]: just run `ament_cpplint` or any of the others in the package's root. It'll spit out things to fix.

**Pull request closed without merging**